src/status_im/contexts/wallet/swap/setup_swap/view.cljs

```diff
@@ -335,12 +335,16 @@
         loading-swap-proposal?                      (rf/sub [:wallet/swap-loading-swap-proposal?])
         swap-proposal                               (rf/sub [:wallet/swap-proposal-without-fees])
         asset-to-pay                                (rf/sub [:wallet/swap-asset-to-pay])
+        asset-to-pay-with-current-account-balance   (rf/sub [:wallet/token-by-symbol
+                                                             (or (:symbol asset-to-pay)
+                                                                 default-token-symbol)])
         asset-to-receive                            (rf/sub [:wallet/swap-asset-to-receive])
         network                                     (rf/sub [:wallet/swap-network])
+        current-account-address                     (rf/sub [:wallet/current-viewing-account-address])
         pay-input-amount                            (controlled-input/input-value pay-input-state)
         pay-token-decimals                          (:decimals asset-to-pay)
         pay-token-balance-selected-chain            (number/convert-to-whole-number
-                                                     (get-in asset-to-pay
+                                                     (get-in asset-to-pay-with-current-account-balance
                                                              [:balances-per-chain
                                                               (:chain-id network) :raw-balance]
                                                              0)
```

Comment:
some of this should also be moved to subscription(s), otherwise the component becomes difficult to manage. There's a [proposal PR](https://github.com/status-im/status-mobile/pull/21507) that is tackling a similar issue for the send input.

Reply:
Updated :)